### ISSUE TITLE
Crashing with Julia 1.8.3

### ISSUE THREAD

**Issue by erlebach** [krlennon/rude#1]
I am running on a Mac with Ventura OS and M1 chip. When reading the pre-trained weights, I get good results. When I train the network from scratch, I get a strange error when executing `rude.jl` from the folder `Giesekus` using Visual Studio Code:
```
#for k = range(start_at,length(protocols),step=1)
#for k = range(start_at,1,step=1)
k = 1
    loss_fn(θ) = loss_univ([θ; p_system], protocols[1:k], tspans[1:k], σ0, σ12_all, k)
    cb_fun(θ, l) = callback(θ, l, protocols[1:k], tspans[1:k], σ0, σ12_all, k)
    result_univ = DiffEqFlux.sciml_train(loss_fn, θi,
                         AMSGrad(),
                         cb = cb_fun,
                         allow_f_increases = false,
                         maxiters = 5)
                         #maxiters = 200)
    global θi = result_univ.minimizer
    @save "tbnn.bson" θi
#end
```
A stack trace is dumped in the terminal, but then the screen clears, and the error is gone. This is very disconcerting. I then reran the code from the terminal (as opposed to Visual Studio Code). Here is the stack trace, which is quite long, so I attached the file. Does this look like an insufficient memory error? Thanks. 
[stack_trace.txt](https://github.com/krlennon/rude/files/10290846/stack_trace.txt)

**Comment by krlennon** [comment]
Reproduced this issue after updating Julia and all dependencies. The original training script crashes Julia, so it's hard to decipher the stack trace. It references Enzyme, so presumably has something to do with the latest AD in Julia.

The SciML ecosystem in Julia has been evolving quickly, and there were a couple of deprecation warnings before Julia crashed (for `FastChain`/`FastDense` and `sciml_train`). I'm planning to move to Lux.jl for ML eventually, but for now just switched to `Flux.Chain`/`Flux.Dense` (should have a minimal effect on performance). Following the example [here](https://docs.sciml.ai/SciMLSensitivity/stable/optimal_control/feedback_control/), I switched the optimization interface to `Optimization.solve`, but the issue persisted.

Following the same example, I also manually defined the sensitivity algorithm for the ODE solver by passing the argument `sensealg = InterpolatingAdjoint(autojacvec=ReverseDiffVJP(true))` to `solve` within the `ensemble_solve` function, and that fixed the issue. So it seems like it had to do with AD after all.

Repo is updated, and the training scripts should work properly with Julia 1.8.3 now (note also the new dependencies).
